api/service/projects_service_test.go

```diff
@@ -197,12 +213,48 @@ func TestProjectsService_UpdateProject(t *testing.T) {
 				authEnforcer.On("UpdateAuthorization", mock.Anything, *tt.expAuthUpdate).Return(nil)
 			}
 
-			projectsService, err := NewProjectsService(MLFlowTrackingURL, storage, authEnforcer, tt.authEnabled)
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				assert.Equal(t, "POST", r.Method)
+				assert.Equal(t, "application/json", r.Header.Get("Content-Type"))
+
+				var payload map[string]interface{}
+				err := json.NewDecoder(r.Body).Decode(&payload)
+				require.NoError(t, err)
+
+				w.WriteHeader(http.StatusOK)
+				response := map[string]string{
+					"status":  "success",
+					"message": "success-message",
+				}
+				err = json.NewEncoder(w).Encode(response)
+				require.NoError(t, err)
+			}))
+
+			UpdateProjectEndpoint := server.URL
+			UpdateProjectPayloadTemplate := `{
+				"project": "{{.Name}}",
+				"administrators": "{{.Administrators}}",
+				"readers": "{{.Readers}}"
+			}`
+			UpdateProjectResponseTemplate := `{
+				"status": "{{.status}}",
+				"message": "{{.message}}"
+			}`
+
+			projectsService, err := NewProjectsService(
+				MLFlowTrackingURL, storage, authEnforcer, tt.authEnabled,
+				config.UpdateProjectConfig{
+					Endpoint:         UpdateProjectEndpoint,
+					PayloadTemplate:  UpdateProjectPayloadTemplate,
+					ResponseTemplate: UpdateProjectResponseTemplate,
+				},
+			)
 			assert.NoError(t, err)
 
-			res, err := projectsService.UpdateProject(context.Background(), tt.arg)
+			res, responseMessage, err := projectsService.UpdateProject(context.Background(), tt.arg)
 			require.NoError(t, err)
 			require.Equal(t, tt.expResult, res)
+			require.Contains(t, responseMessage, "success-message")
```

Comment:
Ummm, can we not check for the existence of only one element in the payload? We should be ensuring that the entire response looks the same as something that we expect.